src/pages/index.js

```diff
@@ -21,79 +21,188 @@ import {
+    });
+  })
+  .catch((err) => console.error(err));
+
 const cardImageModal = new PopupWithImage("#preview-image-modal");
 cardImageModal.setEventListeners();
 
 function handleImageClick(cardData) {
   cardImageModal.open(cardData);
 }
 
-const addCardPopup = new PopupWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit
+const deleteModalConfirmation = new PopupConfirmation(
+  "#modal-confirm",
+  async (cardId, cardElement) => {
+    try {
+      await api.deleteCard(cardId);
+      cardElement.remove();
+    } catch (err) {
+      console.error(`Error On Card Deletion ${err}`);
+    }
+  }
 );
-addCardPopup.setEventListeners();
+
+function handleDeleteCard(card) {
+  deleteModalConfirmation.setSubmitAction(() => {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.remove();
+        deleteModalConfirmation.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+  deleteModalConfirmation.open();
+}
+
+function handleLikeClick(card) {
+  if (card.setIsLiked) {
+    api
+      .dislikeCard(card._id)
+      .then(() => {
+        card.handleLikeIcon();
+      })
+
+      .catch(console.error);
+  } else {
+    api
+      .likeCard(card._id)
+      .then(() => {
+        card.handleLikeIcon();
+      })
+
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
 
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
-  handleProfileEditSubmit
+  async (data) => {
+    await api.updateUserProfile({
+      name: data.title,
+      about: data.description,
+    });
+    userInfo.setUserInfo({
+      title: data.title,
+      description: data.description,
+    });
+  }
 );
 editProfilePopup.setEventListeners();
 
-function handleProfileEditSubmit(formData) {
-  userInfo.setUserInfo({
-    name: formData.title,
-    description: formData.description,
-  });
-  editProfilePopup.close();
-}
+const editAvatarModal = new PopupWithForm(
+  "#edit-avatar-modal",
+  async (formData) => {
+    const avatarUrl = formData.avatar;
+    const userData = await api.updateAvatar(avatarUrl);
+    userInfo.setUserAvatar(userData.avatar);
+  }
+);
 
-function handleAddCardFormSubmit(inputValues) {
-  console.log(inputValues);
-  const name = inputValues.title;
-  const link = inputValues.Url;
-  const cardData = { name, link };
-  cardListEl.addItem(createCard(cardData));
-  console.log(cardData);
-  addCardPopup.close();
-  addCardFormElement.reset();
-}
+editAvatarButton.addEventListener("click", () => {
+  avatarFormValidator.resetValidation();
+  editAvatarModal.open();
+});
+
+editAvatarModal.setEventListeners();
 
 profileEditButton.addEventListener("click", () => {
   const currentUserInfo = userInfo.getUserInfo();
-  profileTitleInput.value = currentUserInfo.name;
+  profileTitleInput.value = currentUserInfo.title;
   profileDescriptionInput.value = currentUserInfo.description;
+  editFormValidator.resetValidation();
   editProfilePopup.open();
 });
 
```

Comment:
You can call `resetValidation` here for the card modal opening to disable the button

```js

addNewCardButton.addEventListener("click", () => {
  addCardPopup.open();
  addFormValidator.resetValidation();
});

```